packages/core/src/components/ux-choice-container-attribute.ts

```diff
@@ -82,19 +82,16 @@ export class UxChoiceContainerAttribute {
     }
   }
 
-  public valueChanged(newValue: string | string[], oldValue?: string | string[]) {
+  public valueChanged(newValue: string | string[]) {
     if (this.multiple === 'auto') {
       this.multipleChanged(); // call this to ensure isMultiple respect value type
     }
     if (this.isMultiple && typeof newValue === 'string') {
       this.value = [];
-      this.requestProcessValue();
     } else if (!this.isMultiple && Array.isArray(newValue)) {
       this.value = undefined;
-      this.requestProcessValue();
-    } else if (!oldValue || newValue !== oldValue) {
-      this.requestProcessValue();
     }
+    this.requestProcessValue();
   }
```

Comment:
Im a bit confused by the logic inside this method. Can you help give some comments, with regards to the logic how we update and request change? One of the confusion I have is how we updating `value` inside a value change, and also immediately requesting process value.

Reply:
I've updated the PR with comments.

Also because the `requestProcessValue()` is behind a `TaskQueue` then it will only be _processed_ once the value is really set properly.

Reply:
Oh but I see. You probably mean that we could `return` after changing the `value` instead of letting go the method to the end and request the process.... I guess that would be better ?

Reply:
`return` when changing value in valueChanged (https://github.com/aurelia/ux/pull/250/commits/4344363cb65d3d4fd7880c08c00f4be8feca5537)

Reply:
Yes, that is what I meant. Can you try and and see if it works better?

Reply:
Yes I tried. It works fine. Not better as both work but with the `return` it will have slightly better perf.